src/helper/audio/audio.cpp

```diff
@@ -338,18 +343,26 @@ namespace Soundux::Objects
             Globals::gAudio.onSoundProgressed(sound, readFrames);
         }
 
-        if (readFrames <= 0)
+        if (sound->repeat && !sound->shouldSeek && readFrames < frameCount)
         {
-            if (sound->repeat)
+            while (frameCount > 0 && readFrames > 0)
```

Comment:
Does this really have to be in this loop?
Wouldn't calling read_pcm_frames after seeking suffice?

Reply:
Also, doesn't this decode the whole audio?
If so that's really not desirable because large files will take long to decode and decoding the whole file just to loop isn't really a good solution imo

Reply:
Let's assume that for calls to Audio::data_callback(), frameCount == 4410. (0.1 of a second, as it is on my machine). 

The looping code has to take into account three possibilities:
- The sound has a number of samples larger than frameCount; let's say 441030 (a teeny bit over 10 seconds at 44.1khz). The last call to Audio::data_callback() will try and fetch 4410 samples, but only get back 30. As such, it'll hit the while loop, which will rewind the sample once and get the remaining 4380 samples to fill the buffer.
- The sound has a number of samples smaller than frameCount; let's say something unrealistic like 80. Audio::data_callback() will try and fetch 4410 samples, but only get back 80. As such, it'll hit the while loop, which will rewind the sample 55 times to generate enough data to fill the sample buffer. (This is the scenario which would be more expensive, but given the sample size is very small this doesn't seem worth worrying about).
- The sound has a number of samples equally divisible by frameCount. Audio::data_callback() will eventually reach a point where ma_decoder_read_pcm_frames returns 0 samples, so it'll hit the if statement and rewind/try a second time.